packages/matter.js/src/codec/BtpCodec.ts

```diff
@@ -0,0 +1,147 @@
+    endSegmentBit: number,
+    beginSegmentBit: number
+}
+
+export interface BtpPacket {
+    header: BtpHeader,
+    payload: BtpPacketPayload
+}
+
+export const enum BtpHeaderBits {
+    HandshakeBit = 0b01000000,
+    ManagementMsg = 0b00100000,
+    AckMsg = 0b00001000,
+    EndSegment = 0b00000100,
+    BeginSegment = 0b00000001
+}
+
+export const enum BtpOpcode {
+    HandshakeManagementOpcode = 0x6c,
+}
+
+const HANDSHAKE_HEADER = 0b01100101;
+
+export class BtpCodec {
+
+    static decodeBtpHandshakeRequest(data: ByteArray): BtpHandshakeRequest {
+        const reader = new DataReader(data, Endian.Little);
+        return this.decodeRequestPayload(reader);
+    }
+
+    static decodeBtpPacket(data: ByteArray): BtpPacket {
+        const reader = new DataReader(data, Endian.Little);
+
+        return {
+            header: this.decodeHeader(reader),
+            payload: this.decodeBtpPacketPayload(reader)
+        };
+
+    }
+
+    static encodeBtpHandshakeResponse({ version, attMtu, windowSize }: BtpHandshakeResponse): ByteArray {
+        return this.encodeHandshakeResponsePayload({ version, attMtu, windowSize });
+    }
+
+    private static decodeBtpPacketPayload(reader: DataReader<Endian.Little>): BtpPacketPayload {
+
+        const ackNumber = reader.readUInt8();
+        const sequenceNumber = reader.readUInt8();
+        const msgLength = reader.readUInt16();
+        const segmentPayload = reader.getRemainingBytes();
+
+        return { ackNumber, sequenceNumber, msgLength, segmentPayload };
+    }
+
+    private static decodeRequestPayload(reader: DataReader<Endian.Little>): BtpHandshakeRequest {
+
+        const header = reader.readUInt8();
+        if (header !== HANDSHAKE_HEADER) throw new Error("Handshake Error - Incorrect BTP Handhshake Request Headers");
+
+        const opcode = reader.readUInt8();
+        if (opcode !== BtpOpcode.HandshakeManagementOpcode) throw new Error("Handshake Error - Management Opcode is incorrect");
+
+        const ver: number[] = [];
+        let version = reader.readInt8();
+        ver[0] = version & 0xF0;
+        ver[1] = version & 0x0F;
+
+        version = reader.readInt8();
+        ver[2] = version & 0xF0;
+        ver[3] = version & 0x0F;
+
+        version = reader.readInt8();
+        ver[4] = version & 0xF0;
+        ver[5] = version & 0x0F;
+
+        version = reader.readInt8();
+        ver[6] = version & 0xF0;
+        ver[7] = version & 0x0F;
+
+        const versions = ver.filter(v => v !== 0)
+        if (versions.length == 0) throw new Error("Handshake Error - Version is incorrect");
+
+        const attMtu = reader.readUInt16();
+        const clientWindowSize = reader.readUInt8();
+
+        return { versions, attMtu, clientWindowSize };
+    }
+
+    private static decodeHeader(reader: DataReader<Endian.Little>): BtpHeader {
+
+        const headerBits = reader.readUInt8();
+        const handshakeBit = (headerBits & BtpHeaderBits.HandshakeBit) >> 6;
```

Comment:
Why you changed that to be a number? Boolean was better ... or was it maybe any of my comments ?

Reply:
changed to number